.travis.yml

```diff
@@ -2,47 +2,85 @@ sudo: false
 language: ruby
 script: bin/test
 install: bin/bootstrap --path vendor/bundle --without development debug
-rvm:
-  - 1.9.3
-  - 2.0.0
-  - 2.1.10
-  - 2.2.7
-  - 2.3.4
-  - 2.4.1
-  - ruby-head
-  - jruby
-  - jruby-20mode
-  - jruby-21mode
-  - jruby-9.1.12.0
 matrix:
   include:
+    - rvm: ruby-head
+      os: linux
+    - rvm: ruby-head
+      os: osx
+    - rvm: 2.4.1
+      os: linux
+      # TravisCI OSX doesn't work with Ruby 2.4.1 yet (https://github.com/travis-ci/travis-ci/issues/7848)
+    - rvm: 2.4.0
+      os: osx
+    - rvm: 2.3.4
+      os: linux
+    - rvm: 2.3.4
+      os: osx
+    - rvm: 2.2
+      os: linux
+    - rvm: 2.2
+      os: osx
+    - rvm: 2.1
+      os: linux
+    - rvm: 2.1
+      os: osx
+    - rvm: 2.0.0
+      os: linux
+    - rvm: 2.0.0
+      os: osx
+    - rvm: 1.9.3
+      os: linux
+    - rvm: 1.9.3
+      os: osx
+    - rvm: jruby
+      os: linux
+      env:
+        # Travis by default also have "-Dcext.enabled=false" set in
+        # JRUBY_OPTS, but JRuby 9 does not support C extensions at all
+        # so it issues warning that will mess up the sterr checks.
+        - JRUBY_OPTS="--server -Xcompile.invokedynamic=false"
```

Comment:
I've done some experiments locally, and the following combination of environment variables greatly improves test time:

```
export JRUBY_OPTS="--client -Xcompile.invokedynamic=false"
export JAVA_OPTS='-Djruby.compile.mode=OFF -XX:+TieredCompilation -XX:TieredStopAtLevel=1'
```

I've tried this with just `bundle exec cucumber features/01_getting_started_with_aruba/run_commands.feature`, and it improved clock runtime from 1m11s to 34s, while reducing system load.

I found these options at https://gist.github.com/lucasallan/2f9b5c945280d226916c and https://github.com/jruby/jruby/wiki/Improving-startup-time.